$MirOS: ports/lang/egcs/gcc4.4/patches/patch-gcc_config_t-mirbsd,v 1.2 2009/12/06 15:52:18 tg Exp $

	DWARF2 unwind support for OpenBSD/MirBSD, from mgcc

--- gcc/config/t-mirbsd.orig	Sun Dec 20 21:36:34 2009
+++ gcc/config/t-mirbsd	Sun Dec 20 21:55:51 2009
@@ -0,0 +1,24 @@
+# $MirOS: ports/lang/egcs/gcc4.4/patches/patch-gcc_config_t-mirbsd,v 1.2 2009/12/06 15:52:18 tg Exp $
+
+# We don't need GCC's own include files.
+USER_H =
+
+# Use unwind-dw2-fde-openbsd
+# Provide weak dummy POSIX threads functions for exception handling
+LIB2ADDEHSTATIC = $(srcdir)/gthr-gnat.c $(srcdir)/unwind-dw2-fde-openbsd.c \
+  $(srcdir)/unwind-c.c $(srcdir)/unwind-dw2.c $(srcdir)/unwind-sjlj.c
+LIB2ADDEHSHARED = $(LIB2ADDEHSTATIC) $(srcdir)/gthr-posix.c
+LIB2ADDEHDEP = unwind.inc unwind-dw2-fde.h unwind-dw2-fde.c \
+  unwind-dw2-fde-glibc.c gthr-posix.c
+LIB2ADDEH = $(LIB2ADDEHSHARED)
+
+# Compile crtbegin.o and crtend.o (and crtbeginT.o) with
+# pie, and crtbeginS.o and crtendS.o with pic.
+CRTSTUFF_T_COMMON = -fno-stack-protector \
+  -fno-omit-frame-pointer -fno-unit-at-a-time
+CRTSTUFF_T_CFLAGS = ${CRTSTUFF_T_COMMON} -fPIE
+CRTSTUFF_T_CFLAGS_S = ${CRTSTUFF_T_COMMON} -fPIC
+
+# Make the shared libraries know themselves
+SHLIB_SOVERSION = 1.0
+SHLIB_SONAME = libgcc_s.so.$(SHLIB_SOVERSION)
